examples/api/test/material/button_style_button/button_style_button.icon_alignment.0_test.dart

```diff
@@ -0,0 +1,213 @@
+    await tester.tap(iconAlignmentEndCc);
+    await tester.pumpAndSettle();
+    await tester.tap(textDirectionLtrCc);
+    await tester.pumpAndSettle();
+    expect(
+      tester.widget<ElevatedButton>(
+        find.byKey(const Key('ElevatedButton.icon')),
+      ).iconAlignment,
+      IconAlignment.end,
+    );
+    expect(
+      tester.widget<FilledButton>(
+        find.byKey(const Key('FilledButton.icon')),
+      ).iconAlignment,
+      IconAlignment.end,
+    );
+    // expect(
+    //   tester.widget<FilledButton>(
+    //     find.byKey(const Key('FilledButton.tonalIcon')),
+    //   ).iconAlignment,
+    //   IconAlignment.end,
+    // );
+    expect(
+      tester.widget<OutlinedButton>(
+        find.byKey(const Key('OutlinedButton.icon')),
+      ).iconAlignment,
+      IconAlignment.end,
+    );
+    expect(
+      tester.widget<TextButton>(
+        find.byKey(const Key('TextButton.icon')),
+      ).iconAlignment,
+      IconAlignment.end,
+    );
+    expect(
+      tester.widget<Directionality>(
+        find.byKey(const Key('Directionality')),
+      ).textDirection,
+      TextDirection.ltr,
+    );
+
+    // IconAlignment.start & TextDirection.rtl
+    await tester.tap(iconAlignmentStartCc);
+    await tester.pumpAndSettle();
+    await tester.tap(textDirectionRtlCc);
+    await tester.pumpAndSettle();
+    expect(
+      tester.widget<ElevatedButton>(
+        find.byKey(const Key('ElevatedButton.icon')),
+      ).iconAlignment,
+      IconAlignment.start,
+    );
+    expect(
+      tester.widget<FilledButton>(
+        find.byKey(const Key('FilledButton.icon')),
+      ).iconAlignment,
+      IconAlignment.start,
+    );
+    expect(
+      tester.widget<FilledButton>(
+        find.byKey(const Key('FilledButton.tonalIcon')),
+      ).iconAlignment,
+      IconAlignment.start,
+    );
+    expect(
+      tester.widget<OutlinedButton>(
+        find.byKey(const Key('OutlinedButton.icon')),
+      ).iconAlignment,
+      IconAlignment.start,
+    );
+    expect(
+      tester.widget<TextButton>(
+        find.byKey(const Key('TextButton.icon')),
+      ).iconAlignment,
+      IconAlignment.start,
+    );
+    expect(
+      tester.widget<Directionality>(
+        find.byKey(const Key('Directionality')),
+      ).textDirection,
+      TextDirection.rtl,
+    );
+
+    // IconAlignment.end & TextDirection.rtl
+    await tester.tap(iconAlignmentEndCc);
+    await tester.pumpAndSettle();
+    await tester.tap(textDirectionRtlCc);
+    await tester.pumpAndSettle();
+    expect(
+      tester.widget<ElevatedButton>(
+        find.byKey(const Key('ElevatedButton.icon')),
+      ).iconAlignment,
+      IconAlignment.end,
+    );
+    expect(
+      tester.widget<FilledButton>(
+        find.byKey(const Key('FilledButton.icon')),
+      ).iconAlignment,
+      IconAlignment.end,
+    );
+    // expect(
+    //   tester.widget<FilledButton>(
+    //     find.byKey(const Key('FilledButton.tonalIcon')),
```

Comment:
I'll take a look tomorrow

Reply:
Sure thanks a lot! I've a backup in my mind. Instead of checking iconAlignment, I'll assert for the text.

```dart
expect(finder, 'ElevatedButton.icon start ltr');
expect(finder, 'ElevatedButton.icon end rtl');
// Etc...
```